app/src/main/java/com/conupods/IndoorMaps/IndoorBuildingOverlays.java

```diff
@@ -0,0 +1,201 @@
+        mLevelButtons = LevelButtons;
+        mMap.setIndoorEnabled(false);
+
+        mImages.clear();
+
+        //index = 0 is first floor of Hall
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h1));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h2));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h8));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h9));
+
+        //index = 4 is first floor of JMSB
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.mb1));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.mbs2));
+
+        //index = 6 is first floor of LOY_CC
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.cc1));
+
+        //index = 7 is first floor of LOY_VL
+        mImages.add(BitmapDescriptorFactory.fromResource((R.drawable.loy_vl1)));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.loy_vl2));
+
+
+        floorButtonsJmsb =  mLevelButtons.findViewById(R.id.floorButtonsJmsb);
+        floorButtonsHall = mLevelButtons.findViewById(R.id.floorButtonsHall);
+        floorButtonsLOY_VL = mLevelButtons.findViewById(R.id.floorButtonsLOYVL);
+    }
+
+    public void hideLevelButton() {
+        floorButtonsJmsb.setVisibility(View.INVISIBLE);
+        floorButtonsHall.setVisibility(View.INVISIBLE);
+        floorButtonsLOY_VL.setVisibility(View.INVISIBLE);
+    }
+
+    public void showButtonsJMSB() {
+        floorButtonsJmsb.setVisibility(View.VISIBLE);
+    }
+
+    public void showButtonsHALL() {
+        floorButtonsHall.setVisibility(View.VISIBLE);
+    }
+
+    public void showButtonsLOYVL() {
+        floorButtonsLOY_VL.setVisibility(View.VISIBLE);
+    }
+
+    //Might be a better way to hidePOIs?
+    private void hidePOIs(int i) {
+        MapStyleOptions style;
+        switch (i) {
+            case 1:
+                style = new MapStyleOptions("[" +
+                        "  {" +
+                        "    \"featureType\":\"poi.business\"," +
+                        "    \"elementType\":\"all\"," +
+                        "    \"stylers\":[" +
+                        "      {" +
+                        "        \"visibility\":\"off\"" +
+                        "      }" +
+                        "    ]" +
+                        "  }," +
+                        "  {" +
+                        "    \"featureType\":\"transit\"," +
+                        "    \"elementType\":\"all\"," +
+                        "    \"stylers\":[" +
+                        "      {" +
+                        "        \"visibility\":\"off\"" +
+                        "      }" +
+                        "    ]" +
+                        "  }" +
+                        "]");
+                break;
+            default:
+                return;
+        }
+        mMap.setMapStyle(style);
+    }
+
+
+    public void displayOverlayJMSB() {
+        hidePOIs(1);
```

Comment:
A lot of repeated code when you do the display Overlay. Extract it into a function and pass all the ints/floats and the mOverlay. maybe the function would be called initializeOverlay. U check if mOverlay==null then u initialize with the proper values. gotta do this to prevent indoor overlay from blobbing